tool/benchmark.cc

```diff
@@ -2,8 +2,15 @@
 // SPDX-License-Identifier: Apache-2.0 OR ISC
 
 #include "internal.h"
+#include <openssl/opensslv.h>
 
 int main(int argc, char **argv) {
+  unsigned long build_version = OPENSSL_VERSION_NUMBER;
+  unsigned long runtime_version = SSLeay();
```

Comment:
are you just concerned with version numbers within a libcrypto "family" (e.g. compiled against OpenSSL 1.1, linked against OpenSSL 3.0) or are you also concerned about accidentally compiling/linking across "families" (e.g. compiled against OpenSSL 1.1, linked against AWS-LC)? the version number may conflict across "families", so you may want to use `OPENSSL_VERSION_TEXT` and `OPENSSL_version` if cross-family linkage is a concern. we [implemented](https://github.com/awslabs/aws-c-cal/pull/186#discussion_r1612497699) a similar check in the CRT recently.